File: skills/instrument-repo/templates/python/sentry_init.py
"""Sentry initialization with PII scrubbing."""

import os

import sentry_sdk


def init_sentry() -> None:
    """Initialize Sentry SDK. No-op when SENTRY_DSN is unset."""
    dsn = os.environ.get("SENTRY_DSN")
    if not dsn:
        return

    env = os.environ.get("SENTRY_ENVIRONMENT", "development")
    release = os.environ.get("SENTRY_RELEASE")

    sentry_sdk.init(
        dsn=dsn,
        environment=env,
        release=release,
        traces_sample_rate=0.1,
        send_default_pii=False,
        before_send=_scrub_pii,
    )


def _scrub_pii(event, hint):
    """Remove PII from Sentry events."""
    user = event.get("user", {})
    if "email" in user:
        user["email"] = "[REDACTED]"
    if "ip_address" in user:
        del user["ip_address"]
    return event

File: skills/instrument-repo/templates/rust/sentry_init.rs
use sentry::ClientInitGuard;

/// Initialize Sentry. Returns a guard that must be held for the program lifetime.
///
/// ```rust
/// fn main() {
///     let _sentry = sentry_init::init();
///     // ... your program ...
/// }
/// ```
pub fn init() -> Option<ClientInitGuard> {
    let dsn = std::env::var("SENTRY_DSN").ok()?;
    if dsn.is_empty() {
        return None;
    }

    let env = std::env::var("SENTRY_ENVIRONMENT").unwrap_or_else(|_| "development".into());
    let release = std::env::var("SENTRY_RELEASE").ok();

    let guard = sentry::init((
        dsn,
        sentry::ClientOptions {
            environment: Some(env.into()),
            release: release.map(Into::into),
            traces_sample_rate: 0.1,
            send_default_pii: false,
            ..Default::default()
        },
    ));

    Some(guard)
}
